content/swift/concepts/optionals/optionals.md

```diff
@@ -0,0 +1,93 @@
+---
+Title: 'Optionals'
+Description: 'Optionals are types that deal with scenarios involving the potential presence or absence of a value (nil).'
+Subjects:
+  - 'iOS'
+  - 'Mobile Development'
+Tags:
+  - 'Optionals'
+CatalogContent:
+  - 'learn-swift'
+  - 'paths/build-ios-apps-with-swiftui'
+---
+
+Optionals are types that deal with scenarios involving the potential presence or absence of a value (`nil`).
+
+## Syntax
+
+In Swift, a non-optional variable should not be set to `nil`. Rather, the optional type must be defined using the added question mark `?`.
+
+There are two ways to use an optional. Below, a variable named `myVariable` is set to an optional `Type`, `Type?`. `Type` referred to a [Swift data type](https://www.codecademy.com/resources/docs/swift/data-types). It can either be equal to some value with the specified data type or `nil`.
+
+```pseudo
+var myVariable: Type?
+var myVariable: Optional<Type>
+```
+
+**Note:** If `myVariable` is not assigned a value of the specified data type, it will default to `nil`.
+
+## Unwrapping Optionals
+
+Optional types should be unwrapped only if it's certain that a value is present. An [error](https://www.codecademy.com/resources/docs/general/error) will result if an optional without a value is unwrapped. Below are some ways this can be done.
+
+### Nil-Coalescing
+
+The nil-coalescing operator, `??`, assigns a value by checking from left to right. If the preceding optional values are `nil`, the default value on the right is assigned. In the example below, since `maybeInt` was previously declared as `nil`, `myInt` will default to `27`:
+
+```swift
+var myInt = maybeInt ?? 27
+```
+
+**Note:** In the following code blocks, a mutable `var` variable can be used in place of any constant `let` variable.
+
+### `if let`
+
+`if let` is an optional binding control structure.
+
+```swift
+var maybeString: String? = "maybeString"
+
+if let myString = maybeString {
+  print("\(maybeString) and myString are equal!")
+} else {
+  print("myString could not be assigned to nil.")
+}
+
+// Output: Optional("maybeString") and myString are equal!
+```
+
+`myString` is declared and conditionally bound to the `maybeString` optional. The value of `myString` can be accessed from inside the `if` statement and within `myString`'s outer scope.
+
+### `guard let`
+
+`guard let` is another optional binding control structure that can be used within a function.
+
+```swift
+guard let myInt = maybeInt else {
+  return
+}
+```
+
+If the optional `maybeInt` holds a value, it will be assigned to `myInt`. Otherwise, the `else` block will run and hit the `return` statement.
+
```

Comment:
Can we have a swift (lol) statement of why `guard let` would be used compared to `if let`, right here?

Reply:
@KTom101 Good point, I felt it was a better transition on line 65, right before we go into `guard let`. LMK what you think!

Reply:
@yangc95 I think it works there too!